app/main.py

```diff
@@ -1,8 +1,26 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+        Person.people[name] = self
+
+
+def create_person_list(people: list[dict]) -> list:
+    person_list = []
+
+    for person_data in people:
+        person = Person(name=person_data.get("name"),
+                        age=person_data.get("age"))
+        person_list.append(person)
+
+    for person_data in people:
+        person = Person.people[person_data.get("name")]
+        if "wife" in person_data and person_data.get("wife"):
+            person.wife = Person.people[person_data.get("wife")]
+        if "husband" in person_data and person_data.get("husband"):
+            person.husband = Person.people[person_data.get("husband")]
```

Comment:
The code assumes that the 'wife' and 'husband' keys, if present, will always have valid corresponding entries in the `people` dictionary. This could lead to a `KeyError` if the specified spouse does not exist. Consider adding a check to ensure the spouse exists in the `people` dictionary before assigning.